File: level_1/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    players = {}
    for p in participant:
        if p in players:
            players[p] += 1
        else:
            players[p] = 1

    for c in completion:
        players[c] -= 1

    for p,count in players.items():
        if count != 0:
            answer = p
    return answer

participant = ['leo', 'kiki', 'eden']
completion = ['eden', 'kiki']

print(solution(participant,completion))

File: Java/src/Hash/완주하지못한선수.java
package Hash;

import java.util.Map;
import java.util.HashMap;
public class 완주하지못한선수 {
	public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String,Integer> hm = new HashMap<>();
        
        for(String name: participant) {
            if(hm.get(name) == null) {
                hm.put(name,1);
            }else {
                int value = hm.get(name);
                hm.put(name,value+1);
            }
        }
        
        for(String name: completion) {
            int value = hm.get(name);
            hm.put(name,value - 1);
        }
        
        for(String name: hm.keySet()) {
            if(hm.get(name) != 0) {
                answer = name;
            }
        }
        return answer;
    }
}
